core/trino-main/src/main/java/io/trino/connector/system/jdbc/FilterUtil.java

```diff
@@ -50,6 +50,20 @@ public static QualifiedTablePrefix tablePrefix(String catalog, Optional<String>
         return new QualifiedTablePrefix(catalog);
     }
 
+    public static Optional<QualifiedTablePrefix> tablePrefixIfPossible(String catalog, Optional<String> schema, Optional<String> table)
+    {
+        if (schema.isPresent() && table.isPresent()) {
+            return Optional.of(new QualifiedTablePrefix(catalog, schema.get(), table.get()));
+        }
+        if (schema.isPresent() && table.isEmpty()) {
+            return Optional.of(new QualifiedTablePrefix(catalog, schema.get()));
+        }
+        if (schema.isEmpty() && table.isEmpty()) {
```

Comment:
my IDE warns me: "Condition 'schema.isEmpty()' is always 'true' "

is this the condition you wanted?

Reply:
same as above